src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerHelper.cs

```diff
@@ -91,10 +92,19 @@ public async Task<Intent> GetIntentForInteractiveBrokerRequestAsync(BrokerReques
                     _logger.Info("Android account manager didn't return any results for interactive broker request.");
                 }
 
-                Bundle bundleResult = (Bundle)await result.GetResultAsync(
-                     AccountManagerTimeoutSeconds,
-                     TimeUnit.Seconds)
-                     .ConfigureAwait(false);
+                Bundle bundleResult = null;
+
+                try
+                {
+                    bundleResult = (Bundle)await result.GetResultAsync(
+                         AccountManagerTimeoutSeconds,
+                         TimeUnit.Seconds)
+                         .ConfigureAwait(false);
+                }
+                catch (OperationCanceledException ex)
+                {
+                    _logger.Error("An error occured when trying to communicate with account manager: " + ex.Message);
```

Comment:
Not sure what happens if you ignore this exception? Does the whole thing eventually fallback to using the browser?

I think we should repackage the exception to explain what is really happening...

`OperationCanceledException if the operation was canceled for any reason, including the user canceling the creation process or adding accounts (of this type) has been disabled by policy`

I bet that in 99% of the reported case, the actual issue comes from "disabled by policy".

I believe that for this issue to be resolved, we would have to properly support a different way of intercting with the broker, i.e. bounded service.